File: proddata/data/shortest_completing_word/Python/Driver.py
from Solution import Solution
import sys
import os

file = open(os.path.join(sys.path[0], '../testcases.txt'), 'r')
lines = file.readlines()

isSolutionWrong = False
lineNumber = 1
solution = Solution()
for line in lines[1:]:
    line = line.strip()
    if lineNumber % 2 == 1:
        testCase = line
        pos = 0
        while line[pos] != ',':
            pos += 1
        licensePlate = line[:pos].rstrip()[1:-1].strip()
        words = [s.strip()[1:-1].strip()
                 for s in line[pos + 1:].lstrip()[1:-1].split(',')]
        actualOutput = solution.shortestCompletingWord(licensePlate, words)
    else:
        expectedOutput = line[1:-1].strip()
        if (actualOutput != expectedOutput):
            print("Result: Failed for test case: " + testCase)
            print("Actual Output: ", actualOutput)
            print("Expected Output: ", expectedOutput)
            isSolutionWrong = True
            break
    lineNumber += 1

if not isSolutionWrong:
    print("Result: Success")

File: proddata/data/merge_sorted_array/Go/Driver.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"strconv"
	"strings"
)

// 3
// [1,2,3,0,0,0], 3, [2,5,6], 3
// [1,2,2,3,5,6]
// [1], 1, [], 0
// [1]
// [0], 0, [1], 1
// [1]

func createIntArrayFromInput(input string) []int {
	array := []int{}
	inputAfterRemovingBraces := strings.TrimSpace(input[1 : len(input)-1])
	if inputAfterRemovingBraces != "" {
		arrayWithStringValues := strings.Split(inputAfterRemovingBraces, ",")
		for _, stringValue := range arrayWithStringValues {
			value, _ := strconv.Atoi(strings.TrimSpace(stringValue))
			array = append(array, value)
		}
	}
	return array
}

func areIntArraysEqual(arr1 []int, arr2 []int) bool {
	if len(arr1) != len(arr2) {
		return false
	}
	for ind, ele := range arr1 {
		if ele != arr2[ind] {
			return false
		}
	}
	return true
}

func createStringFromArray(array []int) string {
	if len(array) == 0 {
		return "[]"
	}
	s := "["
	for ind, ele := range array {
		s += strconv.Itoa(ele)
		if ind != len(array)-1 {
			s += ","
		} else {
			s += "]"
		}
	}
	return s
}

func displayErrorMessage(testCase string, actualOutput string, expectedOutput string) {
	fmt.Println("Result: Failed for test case: ", testCase)
	fmt.Println("Actual Output: ", actualOutput)
	fmt.Println("Expected Output: ", expectedOutput)
}

func main() {
	bs, err := ioutil.ReadFile("../testcases.txt")
	if err != nil {
		os.Exit(1)
	}
	lines := strings.Split(string(bs), "\n")

	isSolutionWrong := false
	var nums1 []int
	var testCase string

	for lineNumber, line := range lines[1:] {
		line = strings.TrimSpace(line)
		if lineNumber%2 == 0 {
			testCase = line
			ind := strings.Index(line, "]")
			nums1 = createIntArrayFromInput(line[:ind+1])
			for line[ind] != '[' {
				ind++
			}
			startofArray2 := ind
			for line[ind] != ']' {
				ind++
			}
			nums2 := createIntArrayFromInput(line[startofArray2 : ind+1])
			merge(nums1, len(nums1)-len(nums2), nums2, len(nums2))
		} else {
			expectedOutput := createIntArrayFromInput(line)
			if !areIntArraysEqual(nums1, expectedOutput) {
				displayErrorMessage(testCase, createStringFromArray(nums1), line)
				isSolutionWrong = true
				break
			}
		}
	}
	if !isSolutionWrong {
		fmt.Println("Result: Success")
	}
}
